pipelines/controller/src/main/java/org/openmrs/analytics/PipelineManager.java

```diff
@@ -315,56 +309,73 @@ synchronized void runIncrementalPipeline()
   /**
    * This method checks upon Controller start checks on Thrift sever to create resource tables if
    * they don't exist.
-   *
-   * @throws IOException
-   * @throws SQLException
    */
-  @PostConstruct
-  private void createResourceTablesOnStart() throws IOException, SQLException {
+  @Override
+  public void onApplicationEvent(ApplicationReadyEvent applicationReadyEvent) {
     if (!dataProperties.isCreateHiveResourceTables()) {
       return;
     }
 
-    DatabaseConfiguration dbConfig =
-        DatabaseConfiguration.createConfigFromFile(dataProperties.getThriftserverHiveConfig());
-
+    DatabaseConfiguration dbConfig;
+    try {
+      dbConfig =
+          DatabaseConfiguration.createConfigFromFile(dataProperties.getThriftserverHiveConfig());
+    } catch (IOException e) {
+      logger.error("Exception while reading thrift hive config.");
+      throw new RuntimeException(e);
+    }
     HiveTableManager hiveTableManager =
         new HiveTableManager(
             dbConfig.makeJdbsUrlFromConfig(),
             dbConfig.getDatabaseUser(),
             dbConfig.getDatabasePassword());
 
-    // Traverse through directories in parquet output directory to collect their names.
-    List<Path> paths;
-    try (Stream<Path> walk =
-        Files.walk(Paths.get(Constants.THRIFT_CONTAINER_PARQUET_PATH_PREFIX))) {
-      paths = walk.filter(Files::isDirectory).collect(Collectors.toList());
-    }
-    Map<String, List<String>> snapshots = new HashMap<>();
-    for (Path path : paths) {
-      String[] tokens = path.toString().split("/");
-      String resource = tokens[tokens.length - 1];
-      if (tokens.length > 0 && dataProperties.getResourceList().indexOf(resource) != -1) {
-        if (!snapshots.containsKey(resource)) {
-          snapshots.put(resource, new ArrayList<>());
-        }
-        snapshots.get(resource).add(path.toString());
+    String rootPrefix = dataProperties.getDwhRootPrefix();
+    Preconditions.checkState(rootPrefix != null && !rootPrefix.isEmpty());
 
-        // Get timestamp from directory name.
-        String[] snapshotsToken = tokens[tokens.length - 2].split(DataProperties.TIMESTAMP_PREFIX);
-        hiveTableManager.createResourceTable(resource, snapshotsToken[1], path.toString());
-      }
-    }
+    String prefix = getPrefix(rootPrefix);
+    String baseDir = getBaseDir(rootPrefix);
+    List<ResourceId> paths;
```

Comment:
nit: Can you move this to where this is initialized on line 340? It seems you don't need this outside `try`.

Reply:
done